### PR TITLE
Bugfix/sleep smartlier 8

### LINKED ISSUE #8: Sleep smartlier
At the moment the scheduler sleeps for 1s. This is good for checks w/ 1s (or multiple of) as check interval, but not needed if all checks have e.g. 10s (or multiple of) as check interval.

Don't always sleep 1s. Instead calculate for how long to sleep after each output.

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #8

### CODE REVIEW

File: checkScheduler.go [hunk 1]
```diff
@@ -26,13 +27,24 @@ func checkScheduler() {
...
+		}
+	}
+
+	numberStart := 0.0
+	i := 0.0
 
 	for {
-		time.Sleep(time.Second)
-		i++
+		sleepTime := (lowestValue - numberStart) + numberStart
```
Comment (Al2Klimov) [inline checkScheduler.go]: What's the goal of this arithmetic operation?

File: toPrint.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-  "check_disk": 3,
-  "check_load": 5,
-  "check_size": 4,
-  "check_apt": 6,
-  "check_tcp": 1
+  "check_disk": 1,
+  "check_load": 2,
+  "check_size": 3,
+  "check_apt": 1,
+  "check_tcp": 2
```
Comment (Al2Klimov) [inline toPrint.json]: What is this change for?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: toPrint.json (hunk 2 above)
Comment (justinmueller05082000) [reply toPrint.json]: it was changed for test purposes should i undo?

Comment (Al2Klimov) [reply toPrint.json]: yes

File: checkScheduler.go (hunk 1 above)
Comment (Al2Klimov) [reply checkScheduler.go]: *bump*

File: toPrint.json (hunk 2 above)
Comment (Al2Klimov) [reply toPrint.json]: *bump*

File: checkScheduler.go [hunk 3]
```diff
@@ -8,6 +8,9 @@ import (
 	"time"
 )
 
+// TODO Rename variables to clearly identifiable
+// Add, commit, push and create a pull request -> Assignee: alex kliimov
+
```
Comment (Al2Klimov) [inline checkScheduler.go]: Srsly?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: checkScheduler.go (hunk 3 above)
Comment (justinmueller05082000) [reply checkScheduler.go]: im a forgetful person i undo it until next push

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Add a test protocol as in #7.

**Comment by justinmueller05082000** [comment]
fixes #9 

for changed code: (1)
`
timeAbsolute := make(map[string]time.Time)

	for service, value := range prints {
		timeAbsolute[service] = time.Now().Add(time.Duration(float64(time.Second) * value))
	}

	for {
		lowestValue := time.Unix(1<<63-1, 1<<63-1)

		for _, value := range timeAbsolute {
			if value.Before(lowestValue) {
				lowestValue = value
			}
		}

		time.Sleep(lowestValue.Sub(time.Now()))
		for text, value := range timeAbsolute {
			if !time.Now().Before(value) {
				fmt.Printf("%s %s \n", time.Now().Format("Monday 15:04:05 01-02-2006"), text)
				timeAbsolute[text] = value.Add(time.Duration(prints[text] * float64(time.Second)))
			}
		}
	}
}
`

terminal result 1: ( programm ran exactly 20 seconds)
Wednesday 10:28:05 01-08-2020 check_tcp 
Wednesday 10:28:06 01-08-2020 check_tcp 
Wednesday 10:28:07 01-08-2020 check_disk 
Wednesday 10:28:07 01-08-2020 check_tcp 
Wednesday 10:28:08 01-08-2020 check_size 
Wednesday 10:28:08 01-08-2020 check_tcp 
Wednesday 10:28:09 01-08-2020 check_load 
Wednesday 10:28:09 01-08-2020 check_tcp 
Wednesday 10:28:10 01-08-2020 check_disk 
Wednesday 10:28:10 01-08-2020 check_apt 
Wednesday 10:28:10 01-08-2020 check_tcp 
Wednesday 10:28:11 01-08-2020 check_tcp 
Wednesday 10:28:12 01-08-2020 check_size 
Wednesday 10:28:12 01-08-2020 check_tcp 
Wednesday 10:28:13 01-08-2020 check_disk 
Wednesday 10:28:13 01-08-2020 check_tcp 
Wednesday 10:28:14 01-08-2020 check_load 
Wednesday 10:28:14 01-08-2020 check_tcp 
Wednesday 10:28:15 01-08-2020 check_tcp 
Wednesday 10:28:16 01-08-2020 check_disk 
Wednesday 10:28:16 01-08-2020 check_size 
Wednesday 10:28:16 01-08-2020 check_apt 
Wednesday 10:28:16 01-08-2020 check_tcp 
Wednesday 10:28:17 01-08-2020 check_tcp 
Wednesday 10:28:18 01-08-2020 check_tcp 
Wednesday 10:28:19 01-08-2020 check_disk 
Wednesday 10:28:19 01-08-2020 check_load 
Wednesday 10:28:19 01-08-2020 check_tcp 
Wednesday 10:28:20 01-08-2020 check_size 
Wednesday 10:28:20 01-08-2020 check_tcp 
Wednesday 10:28:21 01-08-2020 check_tcp 
Wednesday 10:28:22 01-08-2020 check_disk 
Wednesday 10:28:22 01-08-2020 check_apt 
Wednesday 10:28:22 01-08-2020 check_tcp 
Wednesday 10:28:23 01-08-2020 check_tcp 

for unchanged code: (2)
`
i := 0

	for {
		time.Sleep(time.Second)
		i++
		for text, pause := range prints {
			if i%int(pause) == 0 {
				fmt.Printf("%s %s \n", time.Now().Format("Monday 15:04:05 01-02-2006"), text)
			}
		}
	}
}
`

terminal result 2: ( programm ran exactly 20 seconds)
Wednesday 10:30:03 01-08-2020 check_tcp 
Wednesday 10:30:04 01-08-2020 check_tcp 
Wednesday 10:30:05 01-08-2020 check_disk 
Wednesday 10:30:05 01-08-2020 check_tcp 
Wednesday 10:30:06 01-08-2020 check_size 
Wednesday 10:30:06 01-08-2020 check_tcp 
Wednesday 10:30:07 01-08-2020 check_load 
Wednesday 10:30:07 01-08-2020 check_tcp 
Wednesday 10:30:08 01-08-2020 check_apt 
Wednesday 10:30:08 01-08-2020 check_tcp 
Wednesday 10:30:08 01-08-2020 check_disk 
Wednesday 10:30:09 01-08-2020 check_tcp 
Wednesday 10:30:10 01-08-2020 check_size 
Wednesday 10:30:10 01-08-2020 check_tcp 
Wednesday 10:30:11 01-08-2020 check_tcp 
Wednesday 10:30:11 01-08-2020 check_disk 
Wednesday 10:30:12 01-08-2020 check_load 
Wednesday 10:30:12 01-08-2020 check_tcp 
Wednesday 10:30:13 01-08-2020 check_tcp 
Wednesday 10:30:14 01-08-2020 check_tcp 
Wednesday 10:30:14 01-08-2020 check_disk 
Wednesday 10:30:14 01-08-2020 check_size 
Wednesday 10:30:14 01-08-2020 check_apt 
Wednesday 10:30:15 01-08-2020 check_tcp 
Wednesday 10:30:16 01-08-2020 check_tcp 
Wednesday 10:30:17 01-08-2020 check_tcp 
Wednesday 10:30:17 01-08-2020 check_disk 
Wednesday 10:30:17 01-08-2020 check_load 
Wednesday 10:30:18 01-08-2020 check_size 
Wednesday 10:30:18 01-08-2020 check_tcp 
Wednesday 10:30:19 01-08-2020 check_tcp 
Wednesday 10:30:20 01-08-2020 check_apt 
Wednesday 10:30:20 01-08-2020 check_tcp 
Wednesday 10:30:20 01-08-2020 check_disk 
Wednesday 10:30:21 01-08-2020 check_tcp 

Conclusion:
Both programs do the same except for that in terminal result (1) check_tcp is printed last and any other check_* is printed before check_tcp. In terminal result (2) check_tcp is printed in a random sequence. It sometimes is printed first sometimes second sometimes last and so on. Nevertheless is the sequence the only thing which is different.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin bugfix/sleep-smartlier-8`

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**